src/scripts/main.js

```diff
@@ -1,3 +1,273 @@
+      ).textContent;
+
+      if (tableRows[0].classList.contains('sorted')) {
+        return lettersA.localeCompare(lettersB);
+      } else {
+        return lettersB.localeCompare(lettersA);
+      }
+    });
+  } else {
+    result = [...tableRows].sort((a, b) => {
+      const salaryA = +a
+        .querySelector(`td:nth-child(${childNomer})`)
+        .textContent.replace(/,/g, '')
+        .replace(/\$/, '');
+      const salaryB = +b
+        .querySelector(`td:nth-child(${childNomer})`)
+        .textContent.replace(/,/g, '')
+        .replace(/\$/, '');
+
+      if (tableRows[0].classList.contains('sorted')) {
+        return salaryB - salaryA;
+      } else {
+        return salaryA - salaryB;
+      }
+    });
+  }
+
+  const tbody = document.querySelector('tbody');
+
+  tbody.innerHTML = '';
+
+  result.forEach((row) => {
+    tbody.appendChild(row);
+    row.classList.toggle('sorted');
+  });
+}
+
+const pushNotification = (posTop, posRight, title, description, type) => {
+  const notification = document.createElement('div');
+
+  notification.classList.add(type);
+
+  const notificationTitle = document.createElement('h2');
+
+  notificationTitle.classList.add('title');
+  notificationTitle.textContent = title;
+
+  const notificationMessage = document.createElement('p');
+
+  notificationMessage.textContent = description;
+
+  notification.style.top = `${posTop}px`;
+  notification.style.right = `${posRight}px`;
+
+  notification.setAttribute('data-qa', 'notification');
+
+  notification.appendChild(notificationTitle);
+  notification.appendChild(notificationMessage);
+  document.body.appendChild(notification);
+
+  setTimeout(() => {
+    notification.remove();
+  }, 2000);
+};
+
+const filter = document.querySelectorAll('thead tr th');
+const tableBody = document.querySelector('tbody');
+
+for (let i = 0; i < filter.length; i++) {
+  filter[i].addEventListener('click', () => sortRowsByNumber(i));
+}
+
+tableBody.addEventListener('click', (e) => {
+  const active = e.target.closest('tr');
+
+  document
+    .querySelectorAll('tbody tr')
+    .forEach((x) => x.classList.remove('active'));
+  active.classList.add('active');
+});
+
+const form = document.createElement('form');
+
+const namelabel = document.createElement('label');
+
+namelabel.textContent = 'Name: ';
+
+const nameInput = document.createElement('input');
+
+nameInput.type = 'text';
+nameInput.setAttribute('data-qa', 'name');
+nameInput.required = 'true';
+
+const position = document.createElement('label');
+
+position.textContent = 'position: ';
+
+const positionInput = document.createElement('input');
+
+position.type = 'text';
+positionInput.setAttribute('data-qa', 'position');
+positionInput.required = 'true';
+
+const office = document.createElement('label');
+
+office.textContent = 'office: ';
+
+const officeInput = document.createElement('select');
+
+officeInput.setAttribute('data-qa', 'office');
+officeInput.required = 'true';
```

Comment:
The 'required' attribute should be a boolean. Use 'officeInput.required = true;' instead of 'officeInput.required = 'true';'.